File: 参考/comp9041-454/9041-labonserver/test09/distinct_lines.py
#!/usr/bin/python
import sys
import re
d_l=int(sys.argv[1])
a=set()
i=0
flag=0
while 1:
    try:
        line=raw_input()
    except:
        flag=1
        break
    line=re.sub(r'^\s+','',line)
    line=re.sub(r'\s+$','',line)
    line=re.sub(r' +',' ',line)
    line=line.lower()
    i+=1
    a.add(line)
    if len(a)==d_l:
        break
if flag==0:
    print("{} distinct lines seen after {} lines read.".format(d_l,i))
if flag==1:
    print("End of input reached after {} lines read -  {} different lines not seen.".format(i,d_l))

File: others/comp9041-Q/cs9041/test3/distinct_lines.pl
#!/usr/bin/python3
import sys
num = sys.argv[1]
flag = False
d = dict()
count=0
while True:
	
	line = sys.stdin.readline()
	if line == '': break
	count+=1
	line = line.replace(" ","")
	line = line.lower()
	if line not in d:
		d[line] = 1
	else:
		d[line] += 1
	if len(d) == int(num):
		print("{} distinct lines seen after {} lines read.".format(num,count))			
		flag = True
		break
if not flag:
	print("End of input reached after {} lines read - {} different lines not seen.".format(count,num))
